### PR TITLE
Use the rescue from gem in place of old controller/ExceptionNotifier initializers

### PR DESCRIPTION
**Pull request by joemsak**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -123,6 +123,8 @@ gem 'rails_admin'
 # PostgreSQL database
 gem 'pg'
 
+gem 'openstax_rescue_from', '~> 1.2.0'
```
Comment (jpslav) [inline Gemfile]: update to 1.3.0

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class ApplicationController < ActionController::Base
+  use_openstax_exception_rescue
```
Comment (jpslav) [inline app/controllers/application_controller.rb]: take out of here

File: config/initializers/controllers.rb [hunk 3]
```diff
@@ -7,8 +7,6 @@
   helper ApplicationHelper
   helper OpenStax::Utilities::OsuHelper
 
-  rescue_from Exception, :with => :rescue_from_exception
```
Comment (jpslav) [inline config/initializers/controllers.rb]: put `use_openstax_exception_rescue` here (get application controller and api controller in one shot)

File: config/initializers/rescue_from.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  # config.html_error_template_layout_name = 'application'
+
+  # config.email_prefix = "[#{app_name}] (#{app_env}) "
+  config.sender_address = Rails.application.secrets['exceptions']['sender']
+  config.exception_recipients = Rails.application.secrets['exceptions']['recipients']
+end
+
+OpenStax::Api::V1::ApiController.class_eval do
+  use_openstax_exception_rescue
+end
```
Comment (jpslav) [inline config/initializers/rescue_from.rb]: remove this

File: config/initializers/rescue_from.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  # config.notifier = ExceptionNotifier
+
+  # config.html_error_template_path = 'errors/any'
+  # config.html_error_template_layout_name = 'application'
+
+  # config.email_prefix = "[#{app_name}] (#{app_env}) "
+  config.sender_address = Rails.application.secrets['exceptions']['sender']
+  config.exception_recipients = Rails.application.secrets['exceptions']['recipients']
+end
```
Comment (jpslav) [inline config/initializers/rescue_from.rb]: make this config look pretty much exactly like what I did in tutor-server and exchange.  note in particular that it isn't 'exceptions' in the deployment stuff but 'exception'.

File: config/secrets.yml.example [hunk 6]
```diff
@@ -19,6 +19,10 @@ development:
       secret: <%= ENV["OPENSTAX_ACCOUNTS_SECRET"] %>
       url: <%= ENV["OPENSTAX_ACCOUNTS_URL"] || 'http://localhost:2999' %>
       stub: <%= ENV["OPENSTAX_ACCOUNTS_STUB"] %>
+  exceptions:
```
Comment (jpslav) [inline config/secrets.yml.example]: make all three sections look like what I did in tutor-server and exchange (verbatim)

File: config/initializers/controllers.rb (hunk 3 above)
Comment (joemsak) [inline config/initializers/controllers.rb]: Done

**Pull request merged**